### PR TITLE
JSON Content-Type matcher is more inclusive

### PR DESCRIPTION
**Pull request by galiat**
Supports
- application/json
- application/vnd.api+json
- application/vnd.api+json; supported-ext="bulk,jsonpatch"

the JSON API spec uses `application/vnd.api+json`
http://jsonapi.org/extensions/#extension-negotiation

### CODE REVIEW

**Comment by galiat** [comment]
@oestrich  - this modifies some of your code.

I realized browsing other pull requests, there is another one, trying to solve the same problem by @alexandrz 
https://github.com/zipmark/rspec_api_documentation/pull/236/

**Comment by jakehow** [comment]
@galiat I don't think it makes sense to hardcode this here, there are infinite types of vendor specific content types. Would prefer if this was a configurable array w/ the default value of `application/json`, if the user is using some subset of JSON that they still want to pass to the parser by default, they can add their specific type.

**Comment by galiat** [comment]
@jakehow I'm glad you brought that up. I was considering the same thing, but wasn't sure the best way to set up the configuration.

1)  be verbose
`config.pretty_print_response_as_json_for_content_types = [/application\/json/, /application\/vnd.api+json]`

2) mirror post_body_format. 
`config.response_body_format = Proc.new { |params| params }`
`config.response_body_format = :json`

Are there other options? What do you think?

**Comment by oestrich** [comment]
I definitely like the configuration approach more, @jakehow beat me to it!

I'd mirror post body format, keep response/request linked.

**Comment by galiat** [comment]
@oestrich & @jakehow 
How's this?

I can push up a subsequent PR so that users can specify :json or :xml along with a proc.

What do you think about the default value? If you are ok with *not* formatting based on `application/json` content type, I can simplify things by having the proc & `record_response_body ` only have 1 parameter. Downside is that existing users are probably relying on this autoformatting.

**Comment by jakehow** [comment]
@galiat IMO json == application/json

If you are using a specific subset vendor content type it should have a name other than json.  Maybe `json_api` for example.

**Comment by galiat** [comment]
@jakehow We might be misunderstanding each other. Are the symbols (`:json`, `:xml`) assigned to `post_body_format` tied to the Content-Type? This pr doesn't use symbols to specify formatting anyhow.

**Comment by jakehow** [comment]
@galiat sorry, did not notice code changed... it looks good to me, I misunderstood your comment.

I think we should keep the default behavior to preserve backwards compatibility, it could be expanded to include other standard content types if necessary.  The config option should allow anyone using a non-standard type like JSON-API the flexibility they need.

You probably always want the proc to get the `response_content_type` as it is possible an app may respond with > 1 variation.

**Comment by galiat** [comment]
@jakehow is there anything left that I should do before merge?

File: README.md [hunk 1]
```diff
@@ -138,7 +138,12 @@ RspecApiDocumentation.configure do |config|
 
   # Change how the post body is formatted by default, you can still override by `raw_post`
   # Can be :json, :xml, or a proc that will be passed the params
-  config.post_body_formatter = Proc.new { |params| params }
+  config.response_body_formatter = Proc.new { |params| params }
```
Comment (oestrich) [inline README.md]: Should these be flipped?

Comment (galiat) [inline README.md]: oops - :flushed:  Yeap, I'll fix this.

Comment (galiat) [inline README.md]: @jakehow & @oestrich sorry - I edited the original commit - which might not be too obvious via github notifications. I fixed the mistake in documentation. Is this ok to merge now?

**Comment by oestrich** [comment]
Looks good, thanks!

**Pull request merged**